syntax = "proto3";

package ourpc;

message Terminal {
    string id = 1;
}

message Button {
    string label = 1;
    Terminal point_a = 2;
    Terminal point_b = 3;
}

message Led {
    enum Color {
        UNSPECIFIED = 0;
        RED = 1;
        YELLOW = 2;
        GREEN = 3;
    }
    Color color = 2;
    Terminal point_anode = 3;
    Terminal point_cathode = 4;
}

message VoltageSource {
    enum Voltage {
        UNSPECIFIED = 0;
        LOW = 1;
        HIGH = 2;
    }
    Voltage voltage = 1;
    Terminal point = 2;
}

message Wire {
    repeated Terminal terminals = 1;
}

message Circuit {
    repeated Button buttons = 1;
    repeated Led leds = 2;
    repeated VoltageSource voltage_sources = 3;
    repeated Wire wires = 4;
}
